### PR TITLE
Separate addrindex tx fetch error messages

### PR DESCRIPTION
**Pull request by Roasbeef**
Related to #303. 

* Decouple logging and handling of distinct error messages for cases of
failed tx look up.
* This also fixes a bug wherein the `addrindexer` failed to lookup a tx
(since it didn't exist), but continued since the returned error was
nil.

### CODE REVIEW

File: chainindexer.go [hunk 1]
```diff
@@ -467,9 +469,13 @@ func (a *addrIndexer) indexBlockAddrs(blk *btcutil.Block) (database.BlockAddrInd
 				// Lookup and fetch the referenced output's tx.
 				prevOut := txIn.PreviousOutPoint
 				txList, err := a.server.db.FetchTxBySha(&prevOut.Hash)
-				if err != nil || len(txList) == 0 {
-					adxrLog.Errorf("Couldn't get referenced "+
-						"txOut (%v): %v", prevOut, err)
+				if len(txList) == 0 {
+					return nil, errors.New(fmt.Sprintf("tx: %v does not exist",
```
Comment (jrick) [inline chainindexer.go]: `fmt.Errorf`

Comment (Roasbeef) [inline chainindexer.go]: Fixed.

File: chainindexer.go [hunk 2]
```diff
@@ -239,7 +240,7 @@ func (a *addrIndexer) indexManager() {
 			addrIndex, err := a.indexBlockAddrs(indexJob.blk)
 			if err != nil {
 				adxrLog.Errorf("Unable to index transactions of"+
-					" block %v", err)
+					" block, %v", err)
```
Comment (jrick) [inline chainindexer.go]: I'd prefer if this was a colon for consistency with the rest of the logged errors

File: chainindexer.go [hunk 3]
```diff
@@ -400,7 +401,7 @@ out:
...
```
Comment (jrick) [inline chainindexer.go]: colon here too

File: chainindexer.go [hunk 4]
```diff
@@ -467,9 +468,13 @@ func (a *addrIndexer) indexBlockAddrs(blk *btcutil.Block) (database.BlockAddrInd
 				// Lookup and fetch the referenced output's tx.
 				prevOut := txIn.PreviousOutPoint
 				txList, err := a.server.db.FetchTxBySha(&prevOut.Hash)
-				if err != nil || len(txList) == 0 {
-					adxrLog.Errorf("Couldn't get referenced "+
-						"txOut (%v): %v", prevOut, err)
+				if len(txList) == 0 {
+					return nil, fmt.Errorf("tx: %v does not exist",
```
Comment (jrick) [inline chainindexer.go]: "transaction %v not found".  It might exist on someone else's machine.  Also no colon here.

File: chainindexer.go [hunk 5]
```diff
@@ -467,9 +468,13 @@ func (a *addrIndexer) indexBlockAddrs(blk *btcutil.Block) (database.BlockAddrInd
...
+						prevOut.Hash)
+				}
+				if err != nil {
+					adxrLog.Errorf("Error fetching tx (%v): %v",
```
Comment (jrick) [inline chainindexer.go]: I'd remove these parentheses, it's just extra noise.

File: chainindexer.go (hunk 2 above)
Comment (Roasbeef) [inline chainindexer.go]: Noted. Switched to a colon to maintain consistency.

File: chainindexer.go (hunk 3 above)
Comment (Roasbeef) [inline chainindexer.go]: Fixed.

File: chainindexer.go (hunk 4 above)
Comment (Roasbeef) [inline chainindexer.go]: Good point, changed logging message to your suggestion.

File: chainindexer.go (hunk 5 above)
Comment (Roasbeef) [inline chainindexer.go]: Done.

**Comment by jrick** [comment]
ok

**Comment by Roasbeef** [comment]
Rebased and squashed into single commit.

**Comment by davecgh** [comment]
OK after rebase to latest master.

**Comment by Roasbeef** [comment]
Done.

**Pull request merged**